<a href="https://colab.research.google.com/github/OwenGeeAndre/pROJECTS/blob/main/Simple_Multiple_Linear_Regression.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Multiple Linear Regession

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()

from sklearn.linear_model import LinearRegression

# Loading the Data

In [ ]:
data = pd.read_csv(r'/content/1.02.+Multiple+linear+regression.csv')

In [ ]:
data.head(3)

SAT  Rand 1,2,3   GPA
0  1714           1  2.40
1  1664           3  2.52
2  1760           3  2.54

# Create the Multiple Linear Regression

In [ ]:
x = data.drop(columns="GPA")
y = data["GPA"]

## Training the Model

In [ ]:
reg = LinearRegression()
reg.fit(x,y)

LinearRegression()

In [ ]:
reg.coef_

array([ 0.00165354, -0.00826982])

In [ ]:
reg.intercept_

np.float64(0.29603261264909486)

## Calculating the $R^2$ and Adjusted $R^2$

In [ ]:
reg.score(x,y)

0.4066811952814283

### Formula for Adjusted R-squared

$R^2_{adj.} = 1 - (1-R^2)*\frac{n-1}{n-p-1}$

> n = 84 (number of observations)
>
> p = 2 (number of predictors)

In [ ]:
r2 = reg.score(x,y)
n = x.shape[0]
p = x.shape[1]

In [ ]:
adj_r2 = 1-(1-r2)*(n-1)/(n-p-1)
adj_r2

0.3920313482513401

## Feature Selection: F_regression

>F-regression creates simple linear regressions
of each feature and the dependent variable

In [ ]:
from sklearn.feature_selection import f_regression

In [ ]:
f_values, p_values = f_regression(x,y)
p_values

array([7.19951844e-11, 6.76291372e-01])

In [ ]:
p_values.round(3)

array([0.   , 0.676])

* The $p_{value}$ of `SAT`is 0 and `Rand 1,2,3` is 0.676
* with $p_{value} > 0.05$, `Rand 1,2,3` is not a useful predictor

# Creating a summary table

In [ ]:
reg_summary = pd.DataFrame(data=x.columns.values, columns=["Features"])
reg_summary

Features
0         SAT
1  Rand 1,2,3

In [ ]:
reg_summary["coefficients"] = reg.coef_
reg_summary["p-values"] = p_values.round(3)
reg_summary

Features  coefficients  p-values
0         SAT      0.001654     0.000
1  Rand 1,2,3     -0.008270     0.676